Code/GraphMol/RGroupDecomposition/RGroupData.cpp

```diff
@@ -17,16 +17,16 @@
 
 namespace RDKit {
 
-void RGroupData::add(boost::shared_ptr<ROMol> newMol,
+void RGroupData::add(const ROMOL_SPTR &newMol,
                      const std::vector<int> &rlabel_attachments) {
-  // some fragments can be add multiple times if they are cyclic
-  for (auto &mol : mols) {
-    if (newMol.get() == mol.get()) {
-      return;
-    }
+  // some fragments can be added multiple times if they are cyclic
+  if (std::any_of(mols.begin(), mols.end(), [&newMol](const auto &mol) {
```

Comment:
I find the previous code a lot more clear than this.

Reply:
For me, the main advantage of `std::all_of` and `std::any_of` is that they allow to tell at a glance whether all, any or none (for the `!std::any_of` case) of the container items need to fulfil the condition described in the lambda without having to drill down into the logic of `for` loops.

Reply:
We can meet halfway here with I suppose and use the shared_ptrs == operator.  This reads better than the original.

```
if (std::any_of(mols.begin(), mols.end(),
		        [&newMol](const auto &mol) {
		          return newMol == mol;})
    ) {
  return;
 }
```

Reply:
Ha, you added your suggestion while I was committing the revisions based on your previous remarks. I have just committed this last suggestion.